backend/lcfs/web/api/fuel_code/views.py

```diff
@@ -158,46 +156,47 @@ async def get_fuel_code(
     return await service.get_fuel_code(fuel_code_id)
 
 
+@router.post("/{fuel_code_id}/approve", status_code=status.HTTP_200_OK)
+@view_handler([RoleEnum.GOVERNMENT])
+async def approve_fuel_code(
+    request: Request,
+    fuel_code_id: int,
+    service: FuelCodeServices = Depends(),
+):
+    # TODO: Add Logic that checks if the code has necessary fields for approval
+    return await service.approve_fuel_code(fuel_code_id)
+
+
 @router.put("/{fuel_code_id}", status_code=status.HTTP_200_OK)
-@view_handler(["*"])
+@view_handler([RoleEnum.ANALYST])
 async def update_fuel_code(
     request: Request,
     fuel_code_id: int,
-    fuel_code_data: FuelCodeCreateSchema,
+    fuel_code_data: FuelCodeCreateUpdateSchema,
     service: FuelCodeServices = Depends(),
 ):
     return await service.update_fuel_code(fuel_code_id, fuel_code_data)
 
 
 @router.delete("/{fuel_code_id}", status_code=status.HTTP_200_OK)
-@view_handler(["*"])
+@view_handler([RoleEnum.ANALYST])
 async def delete_fuel_code(
     request: Request, fuel_code_id: int, service: FuelCodeServices = Depends()
 ):
     return await service.delete_fuel_code(fuel_code_id)
 
 
 @router.post(
-    "/save",
+    "/",
```

Comment:
Fuel codes seem like they work differently than our other ag grid tables. I just want to confirm that we're ok diverging from the pattern of the compliance schedule tables.

Reply:
I updated it to a standard RESET pattern, if we don't want to follow that I'm okay adding /save back. Is that what you mean?

Reply:
Updated to combine update/save back into a single endpoint.